File: src/main/java/com/rotainteligente/model/Street.java
package com.rotainteligente.model;

public class Street {
    private String streetName;
    private CongestionLevel congestionLevel;
    private boolean hasAccident;
    private boolean isBlocked;

    public Street(String streetName, CongestionLevel congestionLevel, boolean hasAccident, boolean isBlocked) {
        this.streetName = streetName;
        this.congestionLevel = congestionLevel;
        this.hasAccident = hasAccident;
        this.isBlocked = isBlocked;
    }

    public String getStreetName() {
        return streetName;
    }

    public void setStreetName(String streetName) {
        this.streetName = streetName;
    }

    public CongestionLevel getCongestionLevel() {
        return congestionLevel;
    }

    public void setCongestionLevel(CongestionLevel congestionLevel) {
        this.congestionLevel = congestionLevel;
    }

    public boolean hasAccident() {
        return hasAccident;
    }

    public void setHasAccident(boolean hasAccident) {
        this.hasAccident = hasAccident;
    }

    public boolean isBlocked() {
        return isBlocked;
    }

    public void setBlocked(boolean blocked) {
        isBlocked = blocked;
    }

    @Override
    public String toString() {
        return "Street{" +
               "streetName='" + streetName + '\'' +
               ", congestionLevel=" + congestionLevel +
               ", hasAccident=" + hasAccident +
               ", isBlocked=" + isBlocked +
               '}';
    }
}

File: app/models/street.py
from .congestion_level import CongestionLevel

class Street:
    def __init__(self, street_name: str, congestion_level: CongestionLevel, has_accident: bool, is_blocked: bool):
        self.street_name = street_name
        self.congestion_level = congestion_level
        self.has_accident = has_accident
        self.is_blocked = is_blocked

    def __str__(self):
        return (f"Street(name='{self.street_name}', congestion={self.congestion_level}, "
                f"accident={self.has_accident}, blocked={self.is_blocked})")

    def __repr__(self):
        return self.__str__()
